OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearSolverHomotopy.c

```diff
@@ -2234,18 +2234,28 @@ NLS_SOLVER_STATUS solveHomotopy(DATA *data, threadData_t *threadData, NONLINEAR_
       /* Try to get out of here!!! */
       error_f_sqrd        = vec2NormSqrd(homotopyData->n, homotopyData->f1);
 
-      homotopyData->fJac_f(homotopyData, homotopyData->x0, homotopyData->fJac);
-      vecCopy(homotopyData->n, homotopyData->f1, homotopyData->fJac + homotopyData->n*homotopyData->n);
-      vecCopy(homotopyData->n*homotopyData->m, homotopyData->fJac, homotopyData->fJacx0);
-      if (mixedSystem)
-        memcpy(relationsPreBackup, data->simulationInfo->relations, sizeof(modelica_boolean)*data->modelData->nRelations);
-      /* calculate scaling factor of residuals */
-      matVecMultAbsBB(homotopyData->n, homotopyData->fJac, homotopyData->ones, homotopyData->resScaling);
-      debugVectorDouble(OMC_LOG_NLS_JAC, "residuum scaling:", homotopyData->resScaling, homotopyData->n);
-      scaleMatrixRows(homotopyData->n, homotopyData->m, homotopyData->fJac);
-
-      pos = homotopyData->n;
-      assert = (solveSystemWithTotalPivotSearch(data, homotopyData->n, homotopyData->dy0, homotopyData->fJac, homotopyData->indRow, homotopyData->indCol, &pos, &rank, homotopyData->casualTearingSet) == -1);
+      if (error_f_sqrd < newtonFTol)
```

Comment:
`error_f_sqrd` is the squared error so this should probably be compared to the squared tolerance, like with `ftol_sqrd` in the rest of this file.
```suggestion
      if (error_f_sqrd < newtonFTol*newtonFTol)
```